### DISCUSSION TITLE
Deno/Edge Runtimes

### DISCUSSION THREAD

**Discussion by avin-kavish** [fastify/fastify#4105] [Ideas] [answered]
What would it take to make fastify run on Deno?

And maybe this [one](https://nextjs.org/docs/api-routes/edge-api-routes)?

**Comment by jsumners** [comment, marked as answer]
A complete rewrite. This is addressed in our issues. Please search them.

**Comment by avin-kavish** [comment]
yeah, that's what I thought, honestly makes me wonder whether we needed another runtime. Why couldn't we make node run on edge runtimes.

**Reply by mcollina** [reply]
You can use fly.io and do exactly that.

**Reply by avin-kavish** [reply]
fly.io is more like a multi-region deployment isn't it. I'm referring to cloudflare workers, vercel edge functions etc.

**Comment by KaKi87** [comment]
[Deno now supports Fastify](https://deno.com/blog/npm-on-deno-deploy#fastify) ! :tada: 

CC. @ajile-in, @nsisodiya and @gusanchefullstack from fastify/fastify#2173

**Reply by Uzlopak** [reply]
yeah but unfortunately but how can we run our unit tests with deno to ensure they work properly?

**Reply by KaKi87** [reply]
There's [`Deno.test`](https://deno.land/manual/basics/testing).

And you may [use that with `chai`](https://scribe.rip/deno-the-complete-reference/chai-with-deno-test-9afb56748e0d).

**Reply by Uzlopak** [reply]
ridiculous :D
We will not rewrite all our unit tests from tap to chai, just to check that we are running on deno :D.

We have to do it in a minimal invasive manner. if not we can not say, that fastify is 100% deno compatible.

**Reply by KaKi87** [reply]
I'm sorry I thought you were asking as a Fastify user wanting unit test their endpoints.

Well, firstly, cross-runtime testing frameworks aren't a thing, so what you're asking for doesn't exist, secondly, Deno's Node support's purpose isn't to reach 100% compatibility, but to provide enough to ease adoption.

Aside from that, I know the team isn't interested in actively supporting Deno, which is why I only pinged users and not maintainers.

**Reply by KaKi87** [reply]
In other words, I'm saying that Deno supports Fastify even if Fastify doesn't support Deno and we'll make do with this.

**Reply by Uzlopak** [reply]
In other words, deno claims to run fastify but in reality it is not, as it is not guaranteed that it is working properly.

**Reply by KaKi87** [reply]
In reality it is running fine : since the announcement, I already migrated one project and created one other using Fastify on Deno.
